python/lsst/pipe/tasks/postprocess.py

```diff
@@ -156,7 +157,11 @@ def extend(self, table):
                     raise TypeError(f"Type mismatch on column {name!r}: {out_col.dtype} != {in_col.dtype}.")
                 self.result[name][self.index:next_index] = table[name]
             self.index = next_index
-            self.result.meta = astropy.utils.metadata.merge(self.result.meta, table.meta)
+            # Butler provenance should be stripped on merge. It will be
+            # added by butler on write. No attempt is made here to combine
+            # provenance from multiple input tables.
+            self.result.meta = merge_headers([self.result.meta, table.meta], mode="drop")
```

Comment:
Previously it was merging by taking the first value and warning. This change drops duplicate headers. `first` can be used instead.